src/test/run-make-fulldeps/instrument-coverage/Makefile

```diff
@@ -22,50 +20,85 @@ else
 	LLVM_COV_SECTION_PREFIX=
 endif
 
-# This test makes sure that LLVM coverage maps are genereated in LLVM IR.
+all: $(patsubst %.rs,%,$(wildcard *.rs))
 
-COMMON_FLAGS=-Zinstrument-coverage
-
-all:
-	# Compile the test program with instrumentation, and also generate LLVM IR
-	$(RUSTC) $(COMMON_FLAGS) testprog.rs \
+%: %.rs
```

Comment:
@tmandry @wesleywiser - 

I should (now or in the next PR) break up this test into individual tests, so a failure in one (such as `llvm-filecheck` tests) doesn't prevent other independent tests from running.

I also think I should add a test using `-Zinstrument-coverage` _without_ disabling `-Clink-dead-code` (either explicitly or implicitly).

That test would have to `# ignore-msvc` since we know it fails for that target.

I thought I remembered it working for the `windows-gnu` target, but that isn't tested in CI today; but since @mati865 is planning to enable it (in #75872 ) it will be good to have this additional test confirm that, or codify the exception if it doesn't work on `windows-gnu` with `link-dead-code`.

Are you OK if I make these tests refactorings in a follow-on PR? Or do you think I should do that in this one?

Reply:
I took care of this.